Add wrapper for meme version 4.11.0

gregvonkuster opened the pull request:
Dependency installation recipes:
- meme version 4.11.0
- libxml2 version 2.9.3
- libxslt version 1.1.28

gregvonkuster commented:
The ~/tools/meme directory will include a wrapper for fimo in my next PR, so the .shed.yml is my first attempt at "demultiplexing".

bgruening commented on packages/package_libxslt_1_1_28/.shed.yml:
```diff
@@ -0,0 +1,12 @@
+  Libxslt is the XSLT C library developed for the GNOME project. XSLT itself is a an XML language
+  to define transformation for XML. Libxslt is based on libxml2 the XML C library developed for the
+  GNOME project. It also implements most of the EXSLT set of processor-portable extensions functions
+  and some of Saxon''s evaluate and expressions extensions.
```
Saxon_''_s

bgruening commented on packages/package_libxslt_1_1_28/tool_dependencies.xml:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0"?>
+<tool_dependency>
+  <package name="libxml2" version="2.9.3">
```
indentation

bgruening commented on packages/package_meme_4_11_0/tool_dependencies.xml:
```diff
@@ -0,0 +1,24 @@
+            <actions>
+                <action type="download_by_url">http://meme-suite.org/meme-software/4.11.0/meme_4.11.0.tar.gz</action>
+                <action type="set_environment_for_install">
+                    <repository name="package_libxslt_1_1_28" owner="iuc" prior_installation_required="True">
```
The prior_installation_required="True" need to be in the upper package, isn't it?

bgruening commented on tools/meme/.shed.yml:
```diff
@@ -0,0 +1,21 @@
+  (GOMo), measuring the positional enrichment of sequences for known motifs (CentriMo), and analyzing
+  ChIP-seq and other large datasets (MEME-ChIP).  The MEME Suite is comprised of a collection of tools
+  that work together.
+  name_template: "{{ tool_id }}"
```
here is something wrong with the indention, please have a look at: https://github.com/galaxyproject/tools-iuc/blob/master/tools/khmer/.shed.yml

bgruening commented on tools/meme/meme.xml:
```diff
@@ -0,0 +1,316 @@
+        <test>
+            <param name="input1" value="meme_input_1.fasta" ftype="fasta" dbkey="hg19"/>
+            <param name="options_type_selector" value="basic"/>
+            <param name="non_commercial_use" value="True"/>
```
Greg, can we please add 2 more tests, that at least open up a few conditionals! Thanks!

bgruening commented on tools/meme/meme.xml:
```diff
@@ -0,0 +1,316 @@
+                    <when value="no_x_branch" />
+                    <when value="x_branch">
+                        <param name="bfactor" type="integer" value="3" label="Number of iterations of branching" />
+                        <param name="heapsize" type="integer" value="64" label="Maximum number of heaps to use" />
```
I'm hesitated to say that we should convert most of these parameters to `argument=` to get the enhanced help text for free? What do you think @gregvonkuster? `argument="-heapsize"`

gregvonkuster replied:
@bgruening I've not seen this syntax before.  Do you have an example of a tool that does this?

erasche replied:
@gregvonkuster https://github.com/galaxyproject/tools-iuc/blob/a66391d12ef44596fc88db39c068c660d4b62103/tools/bcftools/bcftools_call.xml it automatically documents the command line parameter name for users in the help text, it's quite useful. Additionally it exposes it via the API so we'll be able to do other neat things in the future.

bgruening commented on tools/meme/meme.xml:
```diff
@@ -0,0 +1,316 @@
+<tool id="meme_meme" name="MEME" version="1.0.0">
```
I would call this version 4.11.0.0 ... but this is my personal preference and no IUC recommendation.

gregvonkuster replied:
@bgruening Why add an additional 0 to the end of the version string?  The actual version string for the meme package itself does not include it.

erasche replied:
@gregvonkuster for whenever we bump the galaxy tool version and retain the same wrapper version. This way users can know it's an updated version of the tool interface. It's pretty common in IUC repos, that or `-gx1` or similar. Your preference. Like @bgruening I use it, but there's no real referendum on whether or not to require it yet.

gregvonkuster replied:
@erasche So is that why the version string for meme 4.10.0 looks like this?
&lt;package name="meme" version="4.10.0_4"&gt;
That version string confused me, and its using an "_" rather than a ".".

bgruening commented:
Thanks Greg! A few more tests would be nice, otherwise this looks awesome!

gregvonkuster commented:
@bgruening @blankenberg This single test was the only one that was in the tool I brought over from the Galaxy core.  I'll need some additional test datasets if additional tests are needed to handle the advanced options.

bgruening commented:
If you have time for this would be wonderful!

gregvonkuster commented:
I'm not sure where to get small datasets that I can use for additional tests.  May @blankenberg could come up with some?

yhoogstrate commented on packages/package_libxml2_2_9_3/tool_dependencies.xml:
```diff
@@ -0,0 +1,20 @@
+    <package name="libxml2" version="2.9.3">
+        <install version="1.0">
+            <actions>
+                <action type="download_by_url">ftp://xmlsoft.org/libxml2/libxml2-2.9.3.tar.gz</action>
```
This is a bit tricky url. I see that after a certain time releases get moved to `ftp://xmlsoft.org/libxml2/old/`.

gregvonkuster replied:
I think the core dev team is hosting packages that have this issue, right?  What is the process for getting this package version hosted?

bgruening replied:
Please create a PR against: https://github.com/galaxyproject/cargo-port

gregvonkuster replied:
@bgruening Thanks!  I've submitted a PR.  Do I need to change the tool_dependencies.xml recipe in some way?

bgruening replied:
This package will soon be available in our storage server. Would be nice to include this link and the sha256sum.
This procedure is brand-new and not crystal clear communicated. The aim is to mirror the tarballs and keep them sustainable.

gregvonkuster replied:
@bgruening Sorry, I'm not yet clear how this process works.  My PR for this has been merged, so does that mean the package is now available in the storage server?  How do I get the mirrored URL for this package so I can update my recipe?  Thanks!

gregvonkuster commented:
@bgruening @erasche @blankenberg I've added a 2nd test for the advanced options using a file for the lib parameter that I found here: https://raw.githubusercontent.com/lucapinello/Haystack/master/dependencies/meme_4.9.1/etc/prior30.plib  I hope this make sense since I'm weak on the biology side.  ;)

Also, I believe I have incorporated foxes for everything @bgruening found in hist first review.

Thanks for the help!

gregvonkuster commented:
@bgruening @erasche Is this one ready for merging or is there something else I need to do since I submitted a PR here https://github.com/galaxyproject/cargo-port?  Thanks!

bgruening commented:
@erasche please go to this site and search for meme: https://depot.galaxyproject.org/software/
Than copy the link and us it :)
Thanks a lot!

gregvonkuster commented:
Thanks @bgruening  ;)

martenson commented:
This looks solid to me, the only concern I have right now is the size of the output .html files for tests. 16000 lines is quite a lot, can we somehow make this smaller?
ping @gregvonkuster

mvdbeek commented:
@martenson @gregvonkuster to reduce the file size without changing the tests you could just take something like the last/first 10-20 lines of the output html and in the test add the option `compare="contains"`

mvdbeek commented on tools/meme/meme.xml:
```diff
@@ -0,0 +1,333 @@
+            <param name="options_type_selector" value="advanced"/>
+            <param name="plib" value="prior30.plib" ftype="txt"/>
+            <param name="non_commercial_use" value="True"/>
+            <output name="html_outfile" file="meme_output_html_2.html" lines_diff="12"/>
```
@gregvonkuster 
Here you could use `compare="contains"` to verify that the test result contains a few lines sampled from test-data/meme_output_html_2.html

gregvonkuster replied:
@mvdbeek @martenson Thanks for the info on how to do this.  I've added the fix here: https://github.com/gregvonkuster/tools-iuc/commit/be1e25bc2fa0e00a95d25865f5e52f8762dcaa4e

nekrut commented:
@martenson: this is what we want on main in motif tools section

martenson commented:
@gregvonkuster now we need `git rebase` to get completely rid of the 16000 lines commit, I can do it for you once this is considered 'done'. Is it done from your side?

gregvonkuster commented:
@martenson I've got about 10 "small" things to change based on the reviews, it shouldn't take too long.  I'll ping you when I'm finished.

martenson commented:
yep, no rush

gregvonkuster commented:
@martenson I've submitted a PR here https://github.com/galaxyproject/cargo-port/pull/36, and as soon as I can get the URL changed for the meme 4.11.0 recipe to use the Galaxy depot, this PR can be merged (hopefully).  I'll ping you again when this process is completed.

gregvonkuster commented:
@martenson I believe this latest change https://github.com/galaxyproject/tools-iuc/commit/9f51fca32fc62b363bad2c5f6b69074ebbd0f023 is the final change required by the reviews that have been done so far.

martenson commented:
alright, please give me write access to your repo @gregvonkuster

gregvonkuster commented:
@martenson  I believe you should have it.

martenson commented:
@gregvonkuster thanks, will have a look at this

martenson commented:
Closing in favor of #496 
You can clean your `master` branch now @gregvonkuster
